### PR TITLE
Add a ->to_string() vfunc to FuFirmware

### PR DESCRIPTION
**Pull request by hughsie**
This allows firmware to add to the existing image output and will be used for
future firmware types.

### CODE REVIEW

File: plugins/altos/fu-altos-firmware.c [hunk 1]
```diff
@@ -17,6 +17,12 @@ struct _FuAltosFirmware {
 
 G_DEFINE_TYPE (FuAltosFirmware, fu_altos_firmware, FU_TYPE_FIRMWARE)
 
+static void
+fu_altos_firmware_to_string (FuFirmware *firmware, GString *str)
+{
+	g_string_append (str, "FuAltosFirmware:\n");
```
Comment (superm1) [inline plugins/altos/fu-altos-firmware.c]: Couldn't this be automatically stubbed with the type of plugin by `FuFirmware`?

Comment (hughsie) [reply plugins/altos/fu-altos-firmware.c]: Doh, yes of course. Will fix.

Comment (hughsie) [reply plugins/altos/fu-altos-firmware.c]: Okay, much less new code now, thanks.

**Comment by superm1** [comment]
LGTM if CI is good

**Pull request merged**